### PR TITLE
fix(cds-popover): interactive elements closing wrapped popover onclick

### LINKED ISSUE #21339: [Bug]: unable to open popover web-component in a single click when wrapped in another component
### Package

@carbon/web-components

### Browser

Chrome

### Package version

v11.99.0

### React version

_No response_

### Description

This is related to the [PR](https://github.com/carbon-design-system/carbon/issues/21110) I have already raised.

Issues:

- The popover currently requires two clicks to reopen, but it should reopen on the first click.
- Clicking on interactive elements inside the popover causes it to close : This behaviour works correctly only the first time, on subsequent interactions, popover closes unexpectedly.

It will be better if you could dispatch a close event when open is set to false, so we can handle it on our end.
For the second issue, clicking on interactive elements inside the popover causes it to close - the root cause seems to be the focusout event in popover.ts, where this.open is set to false. 

We’re seeing this problem in two components right now:

- [Coachmark](https://ibm-products-web-components.carbondesignsystem.com/?path=/story/components-coachmark--tooltip)
- [StylePicker](https://labs.carbondesignsystem.com/?path=/story/web-components_components-style-picker-single--color)

Please prioritise this asap.

### Reproduction/example

https://stackblitz.com/edit/vitejs-vite-qksdzzmp?file=src%2Fmy-app.ts

### Steps to reproduce

- Toggle the trigger

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Application/PAL

CARBON FOR IBM PRODUCTS, CARBON LABS

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by maradwan26**
Closes #21339

Fixes an issue with `cds-popover` where clicking a button inside of popover-content on a popover used inside shadow-root with slotted elements closes the popover.

Also adds a new event `cds-popover-closed` which fires when the popover closes via focusout or outsideclick. This is aimed to help adopters sync external `open` states with the popover's `open` state.

E.g:

```
 @cds-popover-closed=${() => {
          this.open = false;
 }}
```

### Changelog

**New**

- Added new event `cds-popover-closed` that fires when the popover closes via focusout or outsideclick.
- Updated `focusout` event handling to properly identify when the popover should actually close
    - Added `mousedown` event handling similar to that used in the [React version of this component ](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/Popover/index.tsx#L199-L213) to account for the same edge case where the popover closes when clicking non-focusable elements (e.g. text)  
- Added a test story that contains a copy of the stackblitz reproduction from the issue to view the fix for this change
    - This story should be removed before merging 
- Added tests to cover this change

#### Testing / Reviewing

- Go to `Popover/Test`
- Clicking the `Done` button should not close the popover
- These changes should not cause regressions to the `Default` nor the `Tab Tip` stories

## PR Checklist

As the author of this PR, before marking ready for review, confirm you:

- [X]  Reviewed every line of the diff
- ~~[ ] Updated documentation and storybook examples~~
- [X] Wrote passing tests that cover this change
- [X] Addressed any impact on accessibility (a11y)
- [X] Tested for cross-browser consistency
- [X] Validated that this code is ready for review and status checks should pass

### CODE REVIEW

File: packages/web-components/src/components/popover/popover.stories.ts [hunk 1]
```diff
@@ -522,6 +524,103 @@ export const TabTipExperimentalAutoAlign = {
...
+        tabtip=""
+        align="bottom-left"
+        id="popover-one"
+        dropshadow=""
+        ?open=${this.open}
+        @cds-popover-closed=${() => {
+          // eslint-disable-next-line no-console
+          console.log('POPOVER CLOSED');
+          this.open = false;
+        }}>
```
Comment (maradwan26) [inline packages/web-components/src/components/popover/popover.stories.ts]: I added this to the issue reproduction as a usage example of the new event.

**Review by heloiselui — COMMENTED** [review]
Thanks for doing this, just a question, should `_handleFocusOut` check `_lastClickWasInsidePopoverContent` before the `relatedTarget` guard so non-focusable clicks (with `relatedTarget` null) don’t still close the popover? Also, do we want to null-guard `triggerEl` before `path.includes(triggerEl)` / `triggerEl.contains(relatedTarget)` in case the trigger slot is empty?

**Comment by maradwan26** [comment]
@heloiselui 

So while testing I found when the relatedTarget was a non-focusable element inside of the popover-content (e.g `<p>` or something), relatedTarget would then be the popover-content instead of just null. Clicking outside, though, would get me a null relatedTarget. In both cases, the functionality is the same because with putting the `_lastClickWasInsidePopoverContent` check before the relatedTarget guard, the flag will have been reset via the setTimeout when clicking outside so `_handleFocusOut` will reach the closing branch, same with putting the check inside of the guard it'll still reach the closing branch.

Either way, I'll move it outside of the guard for robustness just in case something changes or there's an edge case I'm missing.

Same with null-guarding `triggerEl` I will add it just to be safe.

**Review by heloiselui — APPROVED** [review]
LGTM! 🚀

**Review by kennylam — APPROVED** [review]
